src/main/java/uk/gov/companieshouse/accountsDates/AccountsDates.java

```diff
@@ -0,0 +1,199 @@
+
+		// If the days over an exact month are 1 to 15, then round down (unless thegperiod is less than 1 month, in which case round up).
+		// If the days over are 16 to 31, then round up. e.g. 15 months and 4 days is
+		// shown as 15 months; 15 months and 21 days is shown as 16 months.
+		totalMonthsDiff = (accountsPeriod.getDays() >= 15) ? totalMonthsDiff + 1 : totalMonthsDiff; // + 1 to days to include end date in calculation
+
+		// If the previous and current periods both end in the same year, then the
+		// heading is output as a full date e.g. ‘5 January 2015’ ’31 December 2015’.
+		if (isSameYear) {
+			periodObject.put(PERIOD_START, null);
+			periodObject.put(PERIOD_END, convertStringToDisplayDate(convertDateToString(periodEnd)));
+		}
+
+		// If the accounting period is twelve months (+/- 15 days - period start to
+		// period end)
+		// Then the heading for the balance sheet figures is: ccyy e.g. 2015 2016
+		else if ((totalDaysDiff >= yearRangeMin && totalDaysDiff <= yearRangeMax) && totalMonthsDiff == 12) {
+
+//		else if ((accountsPeriod.getYears()==1 && (accountsPeriod.getDays()<=15)) || (accountsPeriod.getMonths()==11 && (accountsPeriod.getDays()>=15))){
+			periodObject.put(PERIOD_START, null);
+			DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy");
+			periodObject.put(PERIOD_END, periodEnd.format(formatter));
+		}
+
+		// If the accounting period is less than twelve months (<350 days - period start
+		// to period end) or
+		// more than twelve months (>380 days - period start to period end)
+		// Then the heading for the balance sheet figures is 'x' months to dd/mm/ccyy
+		// e.g. 15 months to 31/12/2016
+		else if (totalDaysDiff < yearRangeMin || totalDaysDiff > yearRangeMax) {
+			String monthsDiffString = (totalMonthsDiff == 1) ? " month" : " months";
+			periodObject.put(PERIOD_START, totalMonthsDiff + monthsDiffString);
+			periodObject.put(PERIOD_END, convertStringToDisplayDate(convertDateToString(periodEnd)));
+		}
+
+		return periodObject;
+	}
+
+	public boolean isSameYear(LocalDate date1, LocalDate date2) {
+		DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy");
+		String stringDate1 = date1.format(formatter);
+		String stringDate2 = date2.format(formatter);
+
+		return stringDate1.equals(stringDate2);
+	}
+
+	/**
+	 * Takes a {@link Date} and converts it to a Java 8 {@link LocalDate}
+	 * 
+	 * @param date
+	 * @return {@link LocalDate}
+	 */
+
+	public LocalDate convertDateToLocalDate(Date date) {
+		return date != null ? date.toInstant().atZone(ZoneId.systemDefault()).toLocalDate() : null;
+	}
+
+	public List<LocalDate> getPreviousDays(LocalDate currentDate, int numDays) {
+		List<LocalDate> previousDays = new ArrayList<>();
+
+		for (int i = 1; i <= numDays; i++) {
+			previousDays.add(currentDate.minusDays(i));
+		}
+
+		return previousDays;
+	}
+
+	public List<LocalDate> getFutureDays(LocalDate currentDate, int numDays) {
```

Comment:
This will need a javaDoc to explain what it does, there are 4 dates in this method that we are defining and its a bit confusing which one is which. e.g. current and now, I thought they were both the same :(
Additionally, once we know what it does, we may be able to simplify it by using the Streams API like we have in the above `getPreviousDays` :)

Reply:
resolved in 1b655d2